app/controllers/reviews_controller.rb

```diff
@@ -0,0 +1,64 @@
+class ReviewsController < ApplicationController
+  def index
+    @service = Service.find(params[:service_id])
+    @reviews = @service.reviews.all
+  end
+
+  def edit
+    @review = Review.find(params[:id])
+    if review_owned_by_buyer?
+      @review = Review.find(params[:service_id])
+    else
+      flash[:alert] = "You cannot edit someone else's review."
+      render :index
+    end
+  end
+
+  def update
+    @review = Review.find(params[:id])
+    if @review.update(review_params)
+      flash[:notice] = 'Your review has been updated!'
+      redirect_to_index
+    else
+      flash[:alert] = @review.errors.full_messages
+      render :index
+    end
+  end
+
+  def create
+    @review = Review.create(review_params)
+    @review.service_id = params[:service_id]
+    @review.buyer_id = params[current_buyer.id]
```

Comment:
I think `current_buyer` is not included in the `params` object

Reply:
Got it!